[ARC] Internal compiler error when calling an iterator from an inline proc

This code results in a compiler crash. Found in https://github.com/h3rald/min/. Dependency on `critbits` seems to be mandatory for that crash to happen (or maybe it can be done with some other type which has a pairs iterator)

### Example
```nim
# a.nim
import parser

var val = parseMinValue()
if val.kind == minDictionary:
  echo val

# b.nim
import critbits
type
  MinKind* = enum
    minDictionary
    minBool
  MinValue* = object
    case kind*: MinKind
    of minDictionary:
      symbols: CritBitTree[MinOperator]
    else: discard
  MinOperator = object

# remove this inline pragma to make it compile
proc `$`*(a: MinValue): string {.inline.} =
  case a.kind
  of minDictionary:
    result = "hello" # this one is optional, just so we can see output
    for i in a.symbols.pairs:
      result = "hello"
  else: discard

proc parseMinValue*(): MinValue =
  # or this echo
  echo result
```

### Current Output (koch)
```
Error: internal error: invalid kind for lastOrd(tySequence)
Traceback (most recent call last)
/home/dian/.nim/compiler/nim.nim(117) nim
/home/dian/.nim/compiler/nim.nim(82) handleCmdLine
/home/dian/.nim/compiler/cmdlinehelper.nim(84) loadConfigsAndRunMainCommand
/home/dian/.nim/compiler/main.nim(239) mainCommand
/home/dian/.nim/compiler/main.nim(231) compileToBackend
/home/dian/.nim/compiler/main.nim(101) commandCompileToC
/home/dian/.nim/compiler/modules.nim(143) compileProject
/home/dian/.nim/compiler/modules.nim(84) compileModule
/home/dian/.nim/compiler/passes.nim(210) processModule
/home/dian/.nim/compiler/passes.nim(86) processTopLevelStmt
/home/dian/.nim/compiler/cgen.nim(1917) myProcess
/home/dian/.nim/compiler/cgen.nim(968) genProcBody
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2712) expr
/home/dian/.nim/compiler/ccgstmts.nim(1211) genTryGoto
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2666) expr
/home/dian/.nim/compiler/ccgstmts.nim(442) genIf
/home/dian/.nim/compiler/ccgexprs.nim(2627) expr
/home/dian/.nim/compiler/ccgexprs.nim(2284) genMagicExpr
/home/dian/.nim/compiler/ccgexprs.nim(1060) genEcho
/home/dian/.nim/compiler/cgen.nim(596) initLocExpr
/home/dian/.nim/compiler/ccgexprs.nim(2647) expr
/home/dian/.nim/compiler/ccgexprs.nim(2416) genArrayConstr
/home/dian/.nim/compiler/ccgexprs.nim(2664) expr
/home/dian/.nim/compiler/ccgexprs.nim(2440) genStmtListExpr
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2629) expr
/home/dian/.nim/compiler/ccgcalls.nim(627) genCall
/home/dian/.nim/compiler/ccgcalls.nim(237) genPrefixCall
/home/dian/.nim/compiler/ccgcalls.nim(220) genArg
/home/dian/.nim/compiler/cgen.nim(609) initLocExprSingleUse
/home/dian/.nim/compiler/ccgexprs.nim(2635) expr
/home/dian/.nim/compiler/ccgcalls.nim(627) genCall
/home/dian/.nim/compiler/ccgcalls.nim(250) genPrefixCall
/home/dian/.nim/compiler/cgen.nim(596) initLocExpr
/home/dian/.nim/compiler/ccgexprs.nim(2561) expr
/home/dian/.nim/compiler/cgen.nim(1215) genProc
/home/dian/.nim/compiler/cgen.nim(1127) genProcNoForward
/home/dian/.nim/compiler/cgen.nim(1024) genProcAux
/home/dian/.nim/compiler/cgen.nim(968) genProcBody
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2712) expr
/home/dian/.nim/compiler/ccgstmts.nim(1211) genTryGoto
/home/dian/.nim/compiler/ccgexprs.nim(2690) expr
/home/dian/.nim/compiler/ccgstmts.nim(940) genCase
/home/dian/.nim/compiler/ccgstmts.nim(919) genOrdinalCase
/home/dian/.nim/compiler/ccgstmts.nim(180) exprBlock
/home/dian/.nim/compiler/ccgexprs.nim(2663) expr
/home/dian/.nim/compiler/ccgstmts.nim(627) genBlock
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2663) expr
/home/dian/.nim/compiler/ccgstmts.nim(627) genBlock
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2666) expr
/home/dian/.nim/compiler/ccgstmts.nim(442) genIf
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2440) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2665) expr
/home/dian/.nim/compiler/ccgexprs.nim(2451) genStmtList
/home/dian/.nim/compiler/ccgstmts.nim(1561) genStmts
/home/dian/.nim/compiler/ccgexprs.nim(2700) expr
/home/dian/.nim/compiler/ccgstmts.nim(1554) genAsgn
/home/dian/.nim/compiler/ccgstmts.nim(139) loadInto
/home/dian/.nim/compiler/ccgexprs.nim(2633) expr
/home/dian/.nim/compiler/ccgexprs.nim(2285) genMagicExpr
/home/dian/.nim/compiler/ccgexprs.nim(1458) genArrToSeq
/home/dian/.nim/compiler/types.nim(841) lengthOrd
/home/dian/.nim/compiler/types.nim(802) lastOrd
/home/dian/.nim/compiler/msgs.nim(586) internalError
/home/dian/.nim/compiler/msgs.nim(461) rawMessage
/home/dian/.nim/compiler/msgs.nim(458) rawMessage
/home/dian/.nim/compiler/msgs.nim(380) handleError
/home/dian/.nim/compiler/msgs.nim(370) quit
```

### Expected Output
```
hello
hello
```

```
$ nim -v
Nim Compiler Version 1.3.5 [Linux: amd64]
Compiled at 2020-05-17
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 40ac19572a86b5bfa7b57cf0482ae3a30432176a
active boot switches: -d:release -d:danger
```

I tried removing dependency on ``critbits``, but then I got the C compiler error instead, maybe it's the same as in https://github.com/nim-lang/Nim/issues/14251
```nim
# a.nim
import parser

var val = parseMinValue()
if val.kind == minDictionary:
  echo val

# b.nim
type
  MinKind* = enum
    minDictionary
    minBool
  MinValue* = object
    case kind*: MinKind
    of minDictionary:
      symbols: seq[MinOperator]
    else: discard
  MinOperator = object

# remove this inline pragma to make it compile
proc `$`*(a: MinValue): string {.inline.} =
  case a.kind
  of minDictionary:
    result = "hello"
    for i in a.symbols:
      result = "hello"
  else: discard

proc parseMinValue*(): MinValue =
  # or this echo
  echo result
```
Error:
```
/home/dian/.cache/nim/interpreter_d/@minterpreter.nim.c: In function 'dollar___mndzRz9c9ark3YTFhk3Gp6rgparser':                                                                                                                         
/home/dian/.cache/nim/interpreter_d/@minterpreter.nim.c:192:35: error: 'colontmp_' undeclared (first use in this function)                                                                                                              
  192 |    eq___5AN9bgTikL75Wxcrh6teVow((&colontmp_), a._kind_1.symbols);
```

--- a/tests/arc/dmodule.nim
+++ b/tests/arc/dmodule.nim
@@ -1,0 +1,23 @@
+type
+  MinKind* = enum
+    minDictionary
+    minBool
+  MinValue* = object
+    case kind*: MinKind
+    of minDictionary:
+      symbols: seq[MinOperator]
+    else: discard
+  MinOperator = object
+
+# remove this inline pragma to make it compile
+proc `$`*(a: MinValue): string {.inline.} =
+  case a.kind
+  of minDictionary:
+    result = "hello"
+    for i in a.symbols:
+      result = "hello"
+  else: discard
+
+proc parseMinValue*(): MinValue =
+  # or this echo
+  echo result

--- a/tests/arc/t14383.nim
+++ b/tests/arc/t14383.nim
@@ -1,0 +1,13 @@
+discard """
+  cmd: "nim c --gc:arc $file"
+  output: '''
+hello
+hello
+'''
+"""
+
+import dmodule
+
+var val = parseMinValue()
+if val.kind == minDictionary:
+  echo val

